### DISCUSSION TITLE
Olympus .ORF RAW images: `Input file contains unsupported image format` despite building with `-Dtiff=disabled`

### DISCUSSION THREAD

**Discussion by mertalev** [libvips/libvips#3734] [General]
## Bug report

**Describe the bug**

Since some RAW formats (including Olympus) can be mistaken for TIFF, building libvips without libtiff is a way to ensure these images get forwarded to ImageMagick. But this format doesn't seem to be forwarded correctly despite this and gives the error `Input file contains unsupported image format`. Converting the image directly through the `convert` ImageMagick CLI works. 

**To Reproduce**
1. Use [this](https://raw.pixls.us/getfile.php/246/nice/Olympus%20-%20E-520%20-%2016bit%20(4:3).ORF) image
2. Build libvips with `meson setup build --buildtype=release --libdir=lib -Dintrospection=false -Dtiff=disabled`
3. Install sharp
4. Run the below MCVE
6. See error
```

const sharp = require('sharp');

const input = 'Olympus - E-520 - 16bit (4_3).ORF';
const output = 'output.jpg';

async function test() {
    await sharp(input, { failOn: 'none' }).toFile(output);
}

test()
```

**Expected behavior**

It should forward the image to ImageMagick and succeed.

**Actual behavior**

It gives the error `Input file contains unsupported image format`.

**Environment**
(please complete the following information)
 - OS: Debian Bookworm
 - Vips: 8.14.5

**Additional context**

Output of `sharp.format`:

```
{
  jpeg: {
    id: 'jpeg',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: true, buffer: true, stream: true, alias: [Array] }
  },
  png: {
    id: 'png',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: true, buffer: true, stream: true }
  },
  webp: {
    id: 'webp',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: true, buffer: true, stream: true }
  },
  tiff: {
    id: 'tiff',
    input: { file: false, buffer: false, stream: false },
    output: { file: false, buffer: false, stream: false, alias: [Array] }
  },
  magick: {
    id: 'magick',
    input: { file: true, buffer: true, stream: true },
    output: { file: true, buffer: true, stream: true }
  },
  openslide: {
    id: 'openslide',
    input: { file: false, buffer: false, stream: false },
    output: { file: false, buffer: false, stream: false }
  },
  dz: {
    id: 'dz',
    input: { file: false, buffer: false, stream: false },
    output: { file: true, buffer: true, stream: true }
  },
  ppm: {
    id: 'ppm',
    input: { file: true, buffer: false, stream: false, fileSuffix: [Array] },
    output: { file: true, buffer: false, stream: false }
  },
  fits: {
    id: 'fits',
    input: { file: false, buffer: false, stream: false },
    output: { file: false, buffer: false, stream: false }
  },
  gif: {
    id: 'gif',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: false, buffer: false, stream: false }
  },
  svg: {
    id: 'svg',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: false, buffer: false, stream: false }
  },
  heif: {
    id: 'heif',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: true, buffer: true, stream: true, alias: [Array] }
  },
  pdf: {
    id: 'pdf',
    input: { file: false, buffer: false, stream: false },
    output: { file: false, buffer: false, stream: false }
  },
  vips: {
    id: 'vips',
    input: { file: true, buffer: false, stream: false, fileSuffix: [Array] },
    output: { file: true, buffer: false, stream: false }
  },
  jp2k: {
    id: 'jp2k',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: true, buffer: true, stream: true, alias: [Array] }
  },
  jxl: {
    id: 'jxl',
    input: { file: true, buffer: true, stream: true, fileSuffix: [Array] },
    output: { file: true, buffer: true, stream: true }
  },
  raw: {
    id: 'raw',
    input: { file: false, buffer: true, stream: true },
    output: { file: false, buffer: true, stream: true }
  }
}
```

**Comment by lovell** [comment]
>  Converting the image directly through the convert ImageMagick CLI works.

Does `vips` command line work e.g. `vips copy input.orf output.jpg`?

**Reply by jcupitt** [reply]
You could also try `vips magickload input.orf output.jpg`, just to REALLY force load via imagemagick. As Lovell says, it should work.

**Reply by jrasm91** [reply]
This does not work:
```vips copy input.orf output.jpg```
> VipsForeignLoad: "test.ORF" is not a known file format

This works:
```convert test.ORF output.jpg```
(converts file successfully)

**Reply by jrasm91** [reply]
This also works as well:
`vips magickload test.ORF output.jpg`
(converts file successfully)

**Comment by jcupitt** [comment]
Oh I did a little digging and I see:

```
$ convert Olympus\ -\ E-520\ -\ 16bit\ \(4\ 3\).ORF x.jpg
convert-im6.q16: delegate failed `'ufraw-batch' --silent --create-id=also --out-type=png --out-depth=16 --output='%u.png' '%i'' @ error/delegate.c/InvokeDelegate/1966.
```

So the issue is that imagemagick is trying to use `ufraw-batch` for import, and this command is not available in Ubuntu 23.04 (it's unmaintained). It looks like `dcraw` is the current maintained loader, but this imagemagick will not use it.

I found this:

http://www.kevinludlow.com/blog/Configuring_ImageMagick_RAW_Delegates_with_DCRAW_and_UFRAWBatch/1240258/

Which might help.

**Reply by jrasm91** [reply]
If this is true, do you have any insight into why the vips `copy` command, which delegates the conversion to `imagemagick` behaves differently than when calling `convert` directly? Does `imagemagick` use a different loader in `convert` vs however `vips` delegates to it?

**Reply by mertalev** [reply]
My bad for forgetting to mention, but  we build Imagemagick 7 instead of using the distro version. This uses libraw instead of ufraw-batch, so it doesn't have this issue.

**Comment by kleisauke** [comment]
I just opened PR #3735 for this.
